### PR TITLE
Add validation for non-negative integers #16097

### LINKED ISSUE #16097: no warning with invalid pd.set_option('display.max_colwidth')
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

df = pd.DataFrame(np.array([['foo', 'bar', 'bim', 'uncomfortably long string'], ['horse', 'cow', 'banana', 'apple']]))
df
#       0    1       2                          3
# 0    foo  bar     bim  uncomfortably long string
# 1  horse  cow  banana                      apple

pd.set_option('max_colwidth', 2)
df
#       0    1       2                          3
# 0    foo  bar     bim  uncomfortably long string
# 1  horse  cow  banana                      apple

pd.set_option('max_colwidth', 6)
df
#       0    1      2      3
# 0    foo  bar    bim  un...
# 1  horse  cow  ba...  apple
```
#### Problem description

When I set the max_colwidth option to 3 or fewer it does not function and gives no warning. 
I assume that the ellipses take three characters so it has little meaning if you set the option less than 3. 
I think it should warn the user about the invalid setting or display an ellipse filled DataFrame. 

Maybe the documentation could be updated to show range of 3 to xx.

#### Expected Output

```python
pd.set_option('max_colwidth', 2)
# raise ValueError (or similar) .... 
# Invalid argument 2 specificied the minimum is 4

## OR just respond to the new setting but display a useless DateFrame

df
#       0   1   2   3
# 0    ..  ..  ..  ..
# 1    ..  ..  ..  ..

```

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS  
------------------  
commit: None  
python: 3.6.0.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 42 Stepping 7, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.19.2
nose: 1.3.7
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.25.2
numpy: 1.11.3
scipy: 0.18.1
statsmodels: 0.6.1
xarray: None
IPython: 5.1.0
sphinx: 1.5.1
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: 1.2.0
tables: 3.2.2
numexpr: 2.6.1
matplotlib: 2.0.0
openpyxl: 2.4.1
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.7.2
bs4: 4.5.3
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.1.5
pymysql: None
psycopg2: None
jinja2: 2.9.4
boto: 2.45.0
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by ghost**
- [x] closes #16097
 - [x] tests added / passed
 - [ ] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [x] whatsnew entry: "Added validation for non-negative integers when setting display options."

Adds validation for non-negative integers and adds it to options which cannot be negative (e.g. width).

This is my first contribution and I'll need some help with the gif diff check. I've added pandas-dev/pandas as upstream but getting "fatal: bad revision 'upstream/master'" when attempting the diff. Thanks in advance.

### CODE REVIEW

**Comment by jreback** [comment]
pls add tests for this.

File: pandas/core/config_init.py [hunk 1]
```diff
@@ -346,23 +346,24 @@ def table_schema_cb(key):
...
+    cf.register_option('precision', 6, pc_precision_doc, validator=is_nonnegative_int)
     cf.register_option('float_format', None, float_format_doc,
                        validator=is_one_of_factory([None, is_callable]))
-    cf.register_option('column_space', 12, validator=is_int)
+    cf.register_option('column_space', 12, validator=is_nonnegative_int)
     cf.register_option('max_info_rows', 1690785, pc_max_info_rows_doc,
-                       validator=is_instance_factory((int, type(None))))
+                       validator=is_nonnegative_int)
     cf.register_option('max_rows', 60, pc_max_rows_doc,
-                       validator=is_instance_factory([type(None), int]))
```
Comment (jreback) [inline pandas/core/config_init.py]: so you have changed the behavior here, ``None`` is also valid.

**Comment by gfyoung** [comment]
@fastwalker2k4 : 
> I've added pandas-dev/pandas as upstream but getting "fatal: bad revision 'upstream/master'" when attempting the diff

~~~
git remote add upstream git@github.com:pandas-dev/pandas.git
git fetch upstream
~~~

That should do the trick.

**Comment by jreback** [comment]
closing as stale. if you want to re-visit, pls comment and we can re-open.

**Pull request closed without merging**